### PR TITLE
fix mysql foreign key issue

### PR DESCRIPTION
**Pull request by siyamandayubi**
Unlike SQl Server, by default Mysql sets the foregin key as Restricted which causes a problem,
if you want to delete a document

### CODE REVIEW

**Comment by sebastienros** [comment]
Could you add a unit test that would fail without the fix?

**Comment by siyamandayubi** [comment]
Yep, a UnitTest has been added.

File: test/YesSql.Tests/MySqlTests.cs [hunk 1]
```diff
@@ -37,5 +41,36 @@ protected override void OnCleanDatabase(SchemaBuilder builder, DbTransaction tra
             }
             catch { }
         }
+
+        [Fact]
+        public async Task ForeignKeyOfIndexesMustBe_DeleteCascated()
+        {
+            var configuration = CreateConfiguration();
+            _store = StoreFactory.CreateAsync(configuration).GetAwaiter().GetResult();
```
Comment (sebastienros) [inline test/YesSql.Tests/MySqlTests.cs]: await

File: test/YesSql.Tests/MySqlTests.cs [hunk 2]
```diff
@@ -37,5 +41,36 @@ protected override void OnCleanDatabase(SchemaBuilder builder, DbTransaction tra
...
+                Lastname = "Gates"
+            };
+
+            using (var session = _store.CreateSession())
+            {
+                session.Save(bill);
+            }
+
+            // second store, don't register the index
+            _store = StoreFactory.CreateAsync(configuration).GetAwaiter().GetResult();
```
Comment (sebastienros) [inline test/YesSql.Tests/MySqlTests.cs]: await

File: src/YesSql.Provider.MySql/MySqlDialect.cs [hunk 3]
```diff
@@ -96,6 +96,33 @@ public override string GetDropForeignKeyConstraintString(string name)
             return " drop foreign key " + name;
         }
 
+        public override string GetAddForeignKeyConstraintString(string name, string[] srcColumns, string destTable, string[] destColumns, bool primaryKey)
+        {
+            var res = new StringBuilder(200);
```
Comment (sebastienros) [inline src/YesSql.Provider.MySql/MySqlDialect.cs]: call `base.GetAdd...`

**Comment by siyamandayubi** [comment]
@sebastienros  I applied the changes we talked about it.

**Comment by siyamandayubi** [comment]
@sebastienros  I reverted the password

**Pull request merged**